kayenta-objectstore-configbin/src/main/java/com/netflix/kayenta/configbin/storage/ConfigBinStorageService.java

```diff
@@ -285,27 +285,49 @@ public void deleteObject(String accountName, ObjectType objectType, String objec
       String ownerApp = credentials.getOwnerApp();
       String configType = credentials.getConfigType();
       ConfigBinRemoteService remoteService = credentials.getRemoteService();
-      String jsonBody =
-          AuthenticatedRequest.allowAnonymous(
-              () ->
-                  retry.retry(
-                      () -> remoteService.list(ownerApp, configType), MAX_RETRIES, RETRY_BACKOFF));
-
-      try {
-        List<String> ids =
-            kayentaObjectMapper.readValue(jsonBody, new TypeReference<List<String>>() {});
-
-        if (ids.size() > 0) {
-          return ids.stream().map(i -> metadataFor(credentials, i)).collect(Collectors.toList());
+      List<String> ids = new ArrayList<>();
+      boolean hasNext = true;
+      String pageId = null;
+      while (hasNext) {
+        try {
+          final String currentPage = pageId;
+          String jsonBody =
+              AuthenticatedRequest.allowAnonymous(
+                  () ->
+                      retry.retry(
+                          () -> remoteService.list(ownerApp, configType, currentPage),
+                          MAX_RETRIES,
+                          RETRY_BACKOFF));
+          ConfigBinListResponse response =
+              kayentaObjectMapper.readValue(jsonBody, ConfigBinListResponse.class);
+          hasNext = response.hasNext;
+          pageId = response.nextPageId;
+          ids.addAll(
+              response.nameVersions.stream().map(nv -> nv.configName).collect(Collectors.toList()));
+        } catch (IOException e) {
+          log.error("List failed on path {}: {}", ownerApp, e);
```

Comment:
NIT - Unrelated to your change...I don't think will this actually log the full stack trace. I think this is going to call the toString() on the exception which will give the message rather than printing the stack trace. Stacktrace may be more useful for debugging.

Reply:
I'll fix this up

Reply:
So, surprisingly, it turns out that this line works. It ignores the final {} and outsputs something like ERROR: List failed on path kayenta: {}
(full exception stacktrace here) so I removed the extra {}